cli/lsp/analysis.rs

```diff
@@ -1151,6 +1154,145 @@ impl CodeActionCollection {
       ..Default::default()
     }));
   }
+
+  pub async fn add_source_actions(
+    &mut self,
+    document: &Document,
+    range: &lsp::Range,
+    language_server: &language_server::Inner,
```

Comment:
In the future we need to eventually refactor this so that we're not passing dependencies around via method parameters.